### PR TITLE
docs: library generation example of Geo

### PR DESCRIPTION
**Pull request by suztomo**
Adding an advanced options for client library generation.

### CODE REVIEW

File: generation/new_client/README.md [hunk 1]
```diff
@@ -161,6 +161,26 @@ Please create a pull request:
...
+Maps API to generate a module for such library:
+
+```
+~/google-cloud-java$ python3.9 generation/new_client/new-client.py generate \
+  --api_shortname=maps-routing \
+  --proto-path=google/maps/routing \
+  --name-pretty="Routes API" \
+  --product-docs="https://developers.google.com/maps/documentation/routes" \
+  --api-description="Routes API is the next generation, performance optimized version of the existing Directions API and Distance Matrix API. It helps you find the ideal route from A to Z, calculates ETAs and distances for matrices of origin and destination locations, and also offers new features." \
+  --api-id=routes.googleapis.com \
```
Comment (alicejli) [inline generation/new_client/README.md:178]: what does `api-id` do?

File: generation/new_client/README.md [hunk 2]
```diff
@@ -161,6 +161,26 @@ Please create a pull request:
...
+  --cloud-api=false \
+  --requires-billing=true \
```
Comment (alicejli) [inline generation/new_client/README.md:180]: should `requires-billing` be set in general?

File: generation/new_client/README.md [hunk 3]
```diff
@@ -161,6 +161,26 @@ Please create a pull request:
...
+  --distribution-name="com.google.maps:google-maps-routing"
```
Comment (alicejli) [inline generation/new_client/README.md:181]: what does `distribution-name` map to?

**Review by alicejli — COMMENTED** [review]
I'm a little unclear what some of these extra options map to. Should we assume that we need to use this command for any API that's not within `google/cloud`?

File: generation/new_client/README.md (hunk 2 above)
Comment (suztomo) [reply generation/new_client/README.md:234]: It's default to true.

File: generation/new_client/README.md (hunk 3 above)
Comment (suztomo) [reply generation/new_client/README.md:235]: It's Group ID and artifact ID.

**Comment by suztomo** [comment]
@alicejli Good point. I've enriched the "help" option of the command and added these explanation.

File: generation/new_client/README.md [hunk 4]
```diff
@@ -161,6 +161,80 @@ Please create a pull request:
...
+  --product-docs TEXT             Documentation URL that appears in README.md
+                                  [required]
+  --api-description TEXT          Description that appears in README.md
+                                  [required]
+  --release-level [stable|preview]
+                                  A label that appears in repo-metadata.json.
+                                  The first library generation is always
+                                  'preview'.
+  --transport [grpc|http|both]    A label that appears in repo-metadata.json
+  --language TEXT
```
Comment (alicejli) [inline generation/new_client/README.md:188]: Consider adding short description: "Defaults to 'java'".

File: generation/new_client/README.md [hunk 5]
```diff
@@ -161,6 +161,80 @@ Please create a pull request:
...
+  --distribution-name TEXT        Maven coordinates of the generated library.
+                                  By default it's com.google.cloud:google-
+                                  cloud-<api_shortname>
+  --api-id TEXT                   The value of the apiid parameter used in
+                                  README.md It has link to https://console.clo
+                                  ud.google.com/flows/enableapi?apiid=<api_id>
+  --requires-billing BOOLEAN      Based on this value, README.md explains
```
Comment (alicejli) [inline generation/new_client/README.md:195]: Consider adding: "Defaults to `true`".

File: generation/new_client/README.md [hunk 6]
```diff
@@ -161,6 +161,80 @@ Please create a pull request:
...
+                                  library for 'google/maps/routing/v2', then
+                                  you specify this value as
+                                  'google/maps/routing'  [required]
+  --cloud-api BOOLEAN             If true, the artifact ID of the library is
+                                  'google-cloud-'; otherwise 'google-'
+  --group-id TEXT                 The group ID of the artifact when
+                                  distribution name is not set
+  --owlbot-image TEXT             The owlbot container image used in
+                                  OwlBot.yaml
+  --library-type TEXT             A label that appear in repo-metadata.json to
```
Comment (alicejli) [inline generation/new_client/README.md:212]: Consider adding: "Defaults to `GAPIC_AUTO`".

File: generation/new_client/README.md [hunk 7]
```diff
@@ -161,6 +161,80 @@ Please create a pull request:
...
+                                  README.md  [required]
+  --product-docs TEXT             Documentation URL that appears in README.md
+                                  [required]
+  --api-description TEXT          Description that appears in README.md
+                                  [required]
+  --release-level [stable|preview]
+                                  A label that appears in repo-metadata.json.
+                                  The first library generation is always
+                                  'preview'.
+  --transport [grpc|http|both]    A label that appears in repo-metadata.json
```
Comment (alicejli) [inline generation/new_client/README.md:187]: Consider adding: "Defaults to `grpc`"

**Review by alicejli — APPROVED** [review]
Thank you for updating this documentation Tomo! It makes it much clearer what the different options map to. I've left a couple nits, but feel free to merge and we can update those later. Thank you again!

**Pull request merged**